src/nvim/api/ui.c

```diff
@@ -280,6 +280,10 @@ static void remote_ui_mode_change(UI *ui, int mode)
     ADD(args, STRING_OBJ(cstr_to_string("normal")));
   }
   push_call(ui, "mode_change", args);
+
+  Array args2 = ARRAY_DICT_INIT;
+  ADD(args2, INTEGER_OBJ(cursor_shape));
+  push_call(ui, "cursor_shape", args2);
```

Comment:
Why not include this in the mode_change event above? Can't UI events be extended without breaking clients?